### PR TITLE
Implement GET api/reviews endpoint along with its tests and handle an…

### PR DESCRIPTION
**Pull request by Sanagubari**
…y errors

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -30,15 +31,53 @@ describe("GET /api/categories", () => {
...
+        });
+      });
+  });
+  test("200: return reviews sorted by date in descending order, by default", () => {
+    return request(app)
+      .get("/api/reviews")
+      .expect(200)
+      .then(({ body }) => {
+        const { reviews } = body;
+        const sortedReviews = _.cloneDeep(reviews).sort((a, b) => {
```
Comment (k-huggs) [inline __tests__/app.test.js:65]: Nice way to check if the reviews are sorted, you can also use the package jest-sorted

File: app.js [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+const { handle404, handle500 } = require("./controllers/errors.controller");
+const {
+  getCategories,
+  getReviews,
+} = require("./controllers/catogories.controller");
 
 app.get("/api/categories", getCategories);
+app.get("/api/reviews", getReviews);
 
+app.all("/api/*", handle404);
```
Comment (k-huggs) [inline app.js]: I'd use just * to catch all non specified routes, rather than /api/*

File: models/catogories.model.js [hunk 3]
```diff
@@ -5,9 +5,22 @@ exports.selectAllCategories = () => {
...
-
       return rows;
     });
 };
 
-
+exports.selectAllReviews = () => {
+  return db
+    .query(
+      `SELECT  title, designer, owner, review_img_url, category, reviews.votes, reviews.review_id, reviews.created_at,
```
Comment (k-huggs) [inline models/catogories.model.js:15]: Great query 😄

**Review by k-huggs — APPROVED** [review]
Have a look at some of the comments

File: __tests__/app.test.js (hunk 1 above)
Comment (Sanagubari) [reply __tests__/app.test.js:65]: ooo didnt know that, which one would you say is best to use?

**Pull request merged**